### ISSUE TITLE
"LayerNormKernelImpl" not implemented for 'Half' - CPU

### ISSUE THREAD

**Issue by vpj** [pytorch/pytorch#52291]
## 🐛 Bug

May be this is more of a feature request than a bug.

LayerNorm doesn't work on CPU with half precision (it works on GPU though).

## To Reproduce

Run the following piece of code

```python
import torch

torch.layer_norm(torch.tensor([1., 2., 3.], dtype=torch.float16, device='cpu'), normalized_shape=(3,))
```

## Expected behavior

It should just return the normalized vector

## Environment

`1.7.0+cu101` on Google Colab

**Comment by SsnL** [comment]
iirc cpu half is only meant for data transfer but not for computation.

**Comment by mruberry** [comment]
We could also improve the documentation by listing which dtypes torch.layer_norm() supports, like we do with torch.sub() (https://pytorch.org/docs/master/generated/torch.sub.html#torch.sub), for example.

**Comment by Saqib-diar** [comment]
I have faced the same problem, and here is the solution, first do this inside your launch.py:

    commandline_args = os.environ.get('COMMANDLINE_ARGS', "")
    commandline_args = os.environ.get('COMMANDLINE_ARGS', "--skip-torch-cuda-test --no-half")

If you have done it already and still not working then you need to run the below command

> ./webui.sh --precision full --no-half

**Comment by dgcnz** [comment]
Is this feature planned or even desirable? If it's not too complicated I would like to give it a try

**Comment by ashvardanian** [comment]
This feature must be desirable for a very broad range of PyTorch users and currently forbids us from using `f16` as the default for [UForm](https://github.com/unum-cloud/uform) models distribution. The problem is probably coming from this kernel:

https://github.com/pytorch/pytorch/blob/63dcb5b0f2ef3578e81841fd8a2166e732c0ca99/aten/src/ATen/native/cpu/layer_norm_kernel.cpp#L85-L142

I'm trying to extend the test coverage to debug the issue here:

https://github.com/pytorch/pytorch/blob/5a15cbfa440e87f8439d1f28b3d8df0c51daf088/aten/src/ATen/test/math_kernel_test.cpp#L55-L77

But the build instructions are very hard to follow. They are using outdated CMake patterns and partial builds constantly fail. If someone can recommend the correct build flags, I'll be happy to submit a patch.

```bash
# which other flags should I put in .... ?
cmake -DCMAKE_BUILD_TYPE=Debug \
    -DBUILD_SHARED_LIBS=1 \
    -DBUILD_CUSTOM_PROTOBUF=0 \
    -DBUILD_PYTHON=0 \
    -DUSE_CUDA=0 \
    -DUSE_FBGEMM=0 \
    -DUSE_CUPTI_SO=0 \
    -DUSE_METAL=0 \
    -DUSE_NATIVE_ARCH=1 \
    -DUSE_SYSTEM_LIBS=0 \
    -DBUILD_CAFFE2=1 \
    -DPRINT_CMAKE_DEBUG_INFO=1 \
    -DINTERN_BUILD_ATEN_OPS=1 .... -B build

# what's the target name?
cmake --build build --config Debug --parallel 8 --target .... 
```
